### PR TITLE
[VideoPlayer] Improve forced aspect_ratio

### PR DESCRIPTION
**Pull request by peak3d**
## Description
If aspect ratio is forced to be used from stream (either got from ffmpeg or from DVDManager) we currently reevaluate the width of the display area by height * aspect.

The logic searching matching resolutions in whitelist is that either with or hight must match exactly, and the other value must be <= the resolution part.

Example for the logic: If whitelist contains 1920x1080 resolution, 1920x1032 will match, but 2010x1080 does not match, because width of 2010 is greater than 1920. 

Because of this it matters, how we perform the aspect calculation. This is done by this PR.

As part of this 2 other issues were solved:

- The whitelist search was made a little bit more fuzzy, we now allow matches with 8 pixel bigger size because h264 macroblock size of 16byte lead some encoders to not properly encode to 1080 (which is not devidable by 16)

- Searching for "current resolution" in whitelist is fixed using ScreenWidth instead of Width to not compare apples and pies. Comparing whitelist ScreenWidth with CurrentResolution Width leads to unexpected behaviour for all devices which support limiting GUI size.  

## Motivation and Context
https://github.com/xbmc/xbmc/issues/15285

## How Has This Been Tested?
https://github.com/xbmc/xbmc/issues/15285

## Types of change
- [X] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

### CODE REVIEW

File: xbmc/cores/VideoPlayer/VideoPlayerVideo.cpp [hunk 1]
```diff
@@ -670,7 +670,14 @@ bool CVideoPlayerVideo::ProcessDecoderOutput(double &frametime, double &pts)
...
     // use forced aspect if any
     if (m_fForcedAspectRatio != 0.0f)
+    {
       m_picture.iDisplayWidth = (int) (m_picture.iDisplayHeight * m_fForcedAspectRatio);
+      if (m_picture.iDisplayWidth > m_picture.iWidth)
+      {
+        m_picture.iDisplayWidth =  m_picture.iWidth;
+        m_picture.iDisplayHeight = (int) (m_picture.iDisplayWidth / m_fForcedAspectRatio);
+      }
+    }
```
Comment (fritsch) [inline xbmc/cores/VideoPlayer/VideoPlayerVideo.cpp]: That's the wrong place here. You workaround an issue in Resolution.cpp - video player should not do this workaround.

Comment (Kwiboo) [reply xbmc/cores/VideoPlayer/VideoPlayerVideo.cpp]: This makes VideoPlayer follow the same pattern used in video codecs when they apply aspect ratio, e.g. https://github.com/xbmc/xbmc/blob/master/xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecFFmpeg.cpp#L938-L944

**Comment by peak3d** [comment]
@fritsch calculating Displaywidth / height by aspect are already exactly there.
Question why change the width and not the height in the code? Tis is exactly what be done here

**Comment by fritsch** [comment]
Okay.

**Comment by peak3d** [comment]
jenkins build this please

**Pull request merged**